### PR TITLE
[INLONG-10725][CI] Add UT test workflow for flink 1.18 connectors.

### LINKED ISSUE #10725: [Improve][CI] Add UT test workflow file on flink1.18
### Description

This pr #10629 implement the test env for flink 1.18 connector.

But the workflow not has ci_ut file to test flink 1.18.

So, it need to be implement.

### InLong Component

Other for not specified component

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by XiaoYou201**
[INLONG-10725][CI] Add UT test workflow for flink 1.18 connectors.

Fixes #10725

### Motivation

This pr https://github.com/apache/inlong/pull/10629 implement the test env for flink 1.18 connector.

But the workflow not has ci_ut file to test flink 1.18.

So, it need to be implement.

### Modifications

- Modify ci_ut.yml
- Add ci_ut_flink18.yml

### CODE REVIEW

**Review by vernedeng — DISMISSED** [review]

**Comment by XiaoYou201** [comment]
This script will conflict with format-base on flink 1.15. It will be submit when the conflict was solved.

**Review by luchunliang — DISMISSED** [review]

File: inlong-sort/sort-dist/pom.xml [hunk 1]
```diff
@@ -207,11 +232,11 @@
         <profile>
             <id>v1.18</id>
             <dependencies>
-                <dependency>
-                    <groupId>org.apache.inlong</groupId>
-                    <artifactId>sort-format-json-v1.18</artifactId>
-                    <version>${project.version}</version>
-                </dependency>
+                <!--                <dependency>-->
```
Comment (EMsnap) [inline inlong-sort/sort-dist/pom.xml:235]: please remove comment directly

File: inlong-sort/sort-dist/pom.xml [hunk 2]
```diff
@@ -207,11 +232,11 @@
...
+                <!--                    <groupId>org.apache.inlong</groupId>-->
```
Comment (EMsnap) [inline inlong-sort/sort-dist/pom.xml:236]: please remove comment directly

Comment (XiaoYou201) [reply inlong-sort/sort-dist/pom.xml]: fix it

**Review by EMsnap — DISMISSED** [review]

**Review by vernedeng — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

File: inlong-sort/sort-end-to-end-tests/sort-end-to-end-tests-v1.18/pom.xml [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+        <dependency>
+            <groupId>org.elasticsearch.client</groupId>
+            <artifactId>elasticsearch-rest-client</artifactId>
+            <version>${elasticsearch.version}</version>
+        </dependency>
+
+        <dependency>
+            <groupId>org.apache.flink</groupId>
+            <artifactId>flink-shaded-jackson</artifactId>
+            <version>2.15.3-18.0</version>
```
Comment (aloyszhang) [inline inlong-sort/sort-end-to-end-tests/sort-end-to-end-tests-v1.18/pom.xml:78]: better to add a property

Comment (XiaoYou201) [reply inlong-sort/sort-end-to-end-tests/sort-end-to-end-tests-v1.18/pom.xml]: fix it， thx～

**Review by vernedeng — APPROVED** [review]

**Review by aloyszhang — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**